src/PolyhedralGeometry/PolyhedralFan/properties.jl

```diff
@@ -212,8 +212,22 @@ julia> cones(PF, 2)
 """
 function cones(PF::_FanLikeType, cone_dim::Int)
   l = cone_dim - length(lineality_space(PF))
-  l < 1 && return nothing
-  return SubObjectIterator{Cone{_get_scalar_type(PF)}}(
+  t = Cone{_get_scalar_type(PF)}
+  l < 0 && return _empty_subobjectiterator(t, PF)
+
+  if l == 0
+    if length(lineality_space(PF)) == 0
+      return SubObjectIterator{t}(
+        PF, (_, _, _) -> cone(zeros(Int, ambient_dim(PF))), 1, NamedTuple()
+      )
+    end
+  end
```

Comment:
Please use `positive_hull` instead of `cone`. We should be able to pass that `lineality_space` iterator as additional argument to get a cone with that lineality. And we need to make sure the cone has the same coefficient field as the original fan.
```suggestion
  if l == 0
    return SubObjectIterator{t}(
      PF, (_, _, _) -> positive_hull(coefficient_field(PF), zeros(Int, ambient_dim(PF)), lineality_space(PF)), 1, NamedTuple()
    )
  end
```

Reply:
I think this is incorrect. Consider the fan with three cones: a line on the plane, and two half-planes. Then, `lineality_space` returns a ray (!?), even though the lineality space is actually a line. Taking the positive hull or cone of that gives a half-line, which is not equal to the lineality space, and in fact is not equal to any cone in the fan.

Reply:
> Please use positive_hull instead of cone.

Why is `positive_hull` preferable? I see that `cone` is defined using `positive_hull`.

Reply:
> > Please use positive_hull instead of cone.
> 
> Why is `positive_hull` preferable? I see that `cone` is defined using `positive_hull`.

I think cone was mostly a helper function (unfortunately without `_` prefix) and is undocumented (except for `AffineNormalToricVariety`). The export was probably only added for that case.

`positive_hull` is properly documented, also this makes it clear that this is a primal description.

Reply:
> I think this is incorrect. Consider the fan with three cones: a line on the plane, and two half-planes. Then, `lineality_space` returns a ray (!?), even though the lineality space is actually a line. Taking the positive hull or cone of that gives a half-line, which is not equal to the lineality space, and in fact is not equal to any cone in the fan.

`positive_hull` takes `R::AbstractCollection[RayVector]` and an optional argument `L::AbstractCollection[RayVector]]` to pass generators for the lineality space.
So my suggestion
```julia
positive_hull(coefficient_field(PF), zeros(Int, ambient_dim(PF)), lineality_space(PF))
```
will give the sum of the origin with the linear subspace generated by the `RayVectors` which define the lineality of PF.

Your example should look like this:
```julia
julia> p = convex_hull([[1,0],[0,1]])
Polyhedron in ambient dimension 2

julia> nf = normal_fan(p)
Polyhedral fan in ambient dimension 2

julia> dim(nf)
2

julia> rays(nf)
0-element SubObjectIterator{RayVector{QQFieldElem}}

julia> rays_modulo_lineality(nf)
(rays_modulo_lineality = RayVector{QQFieldElem}[[0, -1], [0, 1]], lineality_basis = RayVector{QQFieldElem}[[1, 1]])

julia> ph = positive_hull(coefficient_field(nf), zeros(Int, ambient_dim(nf)), lineality_space(nf))
Polyhedral cone in ambient dimension 2

julia> dim(ph)
1

julia> rays(ph)
0-element SubObjectIterator{RayVector{QQFieldElem}}

julia> lineality_space(ph)
1-element SubObjectIterator{RayVector{QQFieldElem}}:
 [1, 1]
```

Reply:
Okay, I see you are correct. I was confused by `lineality_space` returning a ray.

I think we should defined `cone` to be an alias of `positive_hull` and define a separate `_cone` if we need it somewhere.